src/lib/viewers/BaseViewer.js

```diff
@@ -300,10 +302,17 @@ class BaseViewer extends EventEmitter {
      * @return {void}
      */
     triggerError(err) {
-        const error =
-            err instanceof PreviewError
-                ? err
-                : new PreviewError(ERROR_CODE.LOAD_VIEWER, __('error_refresh'), {}, err.message);
+        let error;
+
+        if (err instanceof PreviewError) {
+            error = err;
+        } else {
+            let message = err || '';
+            message = typeof message === 'string' ? message : err.message;
```

Comment:
Do we have any code that triggers an error with just a string?

Reply:
I'm adding this in as a guard until we're 100% certain we aren't allowing `triggerError()` to accept a string. I couldn't find any instances where we do, but there are quite a few places that call triggerError(), and I don't want anything to blow up.

Reply:
I'm fine with punishing a bad use of triggerError here. If you're not passing in a PreviewError then you get a generic error load viewer code. 

If we don't have this usage anywhere right now, we should prevent it before anyone starts doing that.

Reply:
That sounds fine by me! I'll remove the string case from this, and always expect an error to be passed in